File: neetcode/problems/leetcode509/solution.js
const fibRec = n => {
	if (n <= 1) return n;
	return fibRec(n - 1) + fibRec(n - 2);
};

const fib = n => {
	if (n <= 1) return n;

	let [a, b, c] = [0, 1, 1];

	for (let i = 3; i <= n; i++) {
		a = b;
		b = c;
		c = a + b;
	}

	return c;
};

File: neetcode/problems/leetcode509/solution.py
class SolutionRec:
    def fib(self, n: int) -> int:
        if n <= 1:
            return n
        return self.fib(n - 1) + self.fib(n - 2)


class Solution:
    def fib(self, n: int) -> int:
        if n <= 1:
            return n
        a = 0
        b = 1
        c = 1

        for i in range(3, n + 1):
            a = b
            b = c
            c = a + b

        return c
